packages/eslint-plugin/lib/rules/require-img-alt.js

```diff
@@ -60,14 +66,73 @@ module.exports = {
         if (node.name !== "img") {
           return;
         }
-        if (!hasAltAttrAndValue(node, substitute)) {
-          context.report({
-            loc: {
-              start: node.openStart.loc.start,
-              end: node.openEnd.loc.end,
-            },
-            messageId: MESSAGE_IDS.MISSING_ALT,
-          });
+
+        const altResult = hasValidAltOrSubstitute(node, substitute);
+        const hasSubstituteOption = substitute.length > 0;
+
+        if (!altResult.hasAnyAlt) {
+          if (hasSubstituteOption) {
+            context.report({
+              loc: {
+                start: node.openStart.loc.start,
+                end: node.openEnd.loc.end,
+              },
+              messageId: MESSAGE_IDS.MISSING_ALT,
+            });
+          } else {
+            context.report({
+              loc: {
+                start: node.openStart.loc.start,
+                end: node.openEnd.loc.end,
+              },
+              messageId: MESSAGE_IDS.MISSING_ALT,
+              suggest: [
+                {
+                  messageId: MESSAGE_IDS.INSERT_ALT,
+                  fix(fixer) {
+                    return fixer.insertTextBefore(node.openEnd, ' alt=""');
+                  },
+                },
+              ],
+            });
```

Comment:
```suggestion
            context.report({
              loc: {
                start: node.openStart.loc.start,
                end: node.openEnd.loc.end,
              },
              messageId: MESSAGE_IDS.MISSING_ALT,
              suggest:  hasSubstituteOption ? null : [
                {
                  messageId: MESSAGE_IDS.INSERT_ALT,
                  fix(fixer) {
                    return fixer.insertTextBefore(node.openEnd, ' alt=""');
                  },
                },
              ],
            });
       
```
All the code is the same except for the suggest part. Why not just use suggest conditionally?